### DISCUSSION TITLE
socket.io with nginx

### DISCUSSION THREAD

**Discussion by monil-ia** [socketio/socket.io#4704] [General]
i want to use express server which contain socket.io  in nginx   and use this in react app which run on 3000 but not able to connect  
can any one help me to find how to connect app proxy server solet.io with react app through nginx

worker_processes auto;
events {
  worker_connections 1024;
}

http {
  sendfile on;
  gzip on;
  include mime.types;

  upstream socketio {
    server 192.168.6.1:3000;
    # Add additional servers for load balancing
  } 

  server {
    listen 80;
    server_name  localhost;

     location /socket.io/ {
      proxy_pass http://socketio;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "Upgrade";
      proxy_set_header Host $host;
      proxy_cache_bypass $http_upgrade;
      proxy_hide_header 'Access-Control-Allow-Origin';
      proxy_hide_header 'Access-Control-Allow-Headers';
      proxy_hide_header 'Access-Control-Allow-Methods';
      # Additional Socket.IO configuration options go here
    }

      location /register/ {
      proxy_pass http://192.168.6.1:5000/;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection 'upgrade';
      proxy_set_header Host $host;
      proxy_cache_bypass $http_upgrade;
      proxy_hide_header 'Access-Control-Allow-Origin';
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      
       
    }

     location /auth/ {
      proxy_pass http://192.168.6.1:5001/;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection 'upgrade';
      proxy_set_header Host $host;
      proxy_cache_bypass $http_upgrade;
      proxy_hide_header 'Access-Control-Allow-Origin';
       proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    }

      location /admin/ {
      proxy_pass http://192.168.6.1:5002/;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection 'upgrade';
      proxy_set_header Host $host;
      proxy_cache_bypass $http_upgrade;
      proxy_hide_header 'Access-Control-Allow-Origin';
       proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    }

  }
} 

i want to use express server which contain socket.io  in nginx   and use this in react app which run on 3000 but not able to connect  
can any one help me to find how to connect app proxy server solet.io with react app through nginx

**Comment by darrachequesne** [comment]
Hi! Did you check the documentation [here](https://socket.io/docs/v4/reverse-proxy/#nginx)?

**Reply by monil-ia** [reply]
upstream socketio {
    server 192.168.6.1:5000;
  server 192.168.6.1:5001;
   server 192.168.6.1:5002;
    # Add additional servers for load balancing
  }   yes able to connect with 5000 only not with other two
